pages/scenario_comparison_page.py

```diff
@@ -19,3 +20,110 @@ def comparing_scenarios_page():
+        )
+
+    st.markdown("<MISSING: TEXT EXPLAINING PLOT BELOW, e.g. the impact of subsidy trajectories and levy rebalancing in the results.>")
+
+    results = pd.read_csv("s3://asf-lifetime-cost-model/outputs/cost_dif_by_year_decile_" + str(cost_decile) + ".csv")
+    results.rename(columns={"Unnamed: 0": "installation_year"}, inplace=True)
+    results = results.melt(
+        id_vars=["installation_year"],
+        var_name="scenario",
+        value_name="cost_difference_ashp_minus_boiler"
+    )    
+    scenarios_list = results["scenario"].unique().tolist()
+
+    st.markdown("As default, the chart shows annualised lifetime costs, i.e. the average yearly cost over the lifetime of the heating system. You can toggle to see total lifetime costs instead.")
+    toggle_total_lifetime_costs = st.toggle(
+        label="Show total lifetime costs instead of annualised lifetime costs",
+        value=False,
+        key="toggle_total_lifetime_costs",
+        help="Toggle between annualised lifetime costs and total lifetime costs",
+    )
+
+    if toggle_total_lifetime_costs:
+        title = ["Total difference between air source heat pumps and gas boilers by installation year and scenario for average household",
+                 "(values below zero indicate air source heat pumps are cheaper)"]
+    else:
+        results["cost_difference_ashp_minus_boiler"] = (
+            results["cost_difference_ashp_minus_boiler"] / 15  # annualise over 15 years
+        )
+        title = ["Annual difference between air source heat pumps and gas boilers by installation year and scenario for average household",
+                 "(values below zero indicate air source heat pumps are cheaper)"]
+    line_chart = (
+        alt.Chart(results)
+        .mark_line(point=True, strokeWidth=3)
+        .encode(
+            x=alt.X("installation_year:O", title="Installation year"),
+            y=alt.Y("cost_difference_ashp_minus_boiler:Q", title=""),
+            color=alt.Color("scenario:N", title="Scenario", scale=alt.Scale(domain=scenarios_list, range=NESTA_COLOURS)),
+            tooltip=[
+                alt.Tooltip("installation_year:O", title="Installation year"),
+                alt.Tooltip("scenario:N", title="Scenario"),
+                alt.Tooltip("cost_difference_ashp_minus_boiler:Q", title="Cost difference (heat pump - boiler)", format=".2f"),
+            ],
+        )
+    )
+
+    # Shaded area for y < 0
+    min_value = results["cost_difference_ashp_minus_boiler"].min()
+
+    # Create dummy DataFrame covering the x-axis
+    area_df = pd.DataFrame({
+        "installation_year": results["installation_year"].unique(),
+        "y": [min_value] * len(results["installation_year"].unique())
+    })
+
+    area_below_zero = (
+        alt.Chart(area_df)
+        .mark_area(color="lightgray", opacity=0.3)
+        .encode(
+            x=alt.X("installation_year:O", title="Installation year"),
```

Comment:
```suggestion
            x=alt.X("installation_year:O"),
```

For some reason, defining the x-axis title for this chart layer as well means that there is no x-axis label in the final combined chart. I tested removing it here and it managed to bring the x-axis label back :)